recipes/libftdi/1.x/conanfile.py

```diff
@@ -0,0 +1,101 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibFtdiConan(ConanFile):
+    name = "libftdi"
+    description = "A library to talk to FTDI chips"
+    license = ["GNU LGPL v2.1"]
+    topics = ("conan", "libftdi1")
+    homepage = "https://www.intra2net.com/en/developer/libftdi/"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package", "pkg_config"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    requires = (
+            "libusb/1.0.23",
+            "boost/1.74.0"
+            )
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "libftdi1-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos is not supported")
+        if self.settings.compiler == "Visual Studio":
+            raise ConanInvalidConfiguration("Building with Visual Studio is not supported")
+
+    def build_requirements(self):
+        if not tools.which("pkg-config"):
+            self.build_requires("pkgconf/1.7.3")
+
+    def _patch_cmakelists(self, subfolder):
+        cmakelists_path = os.path.join(self._source_subfolder, subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists_path, "CMAKE_SOURCE_DIR", "PROJECT_SOURCE_DIR", strict=False)
+        tools.replace_in_file(cmakelists_path, "CMAKE_BINARY_DIR", "PROJECT_BINARY_DIR", strict=False)
```

Comment:
Looking over the project. I think we can do sometime much less aggressive. 

Can you please try to add the following? This way we can contribute it back to upstream 🤞 

[cmake-allow-subproject.patch.txt](https://github.com/conan-io/conan-center-index/files/5743632/cmake-allow-subproject.patch.txt)